dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/DataOwnerApi.kt

```diff
@@ -0,0 +1,129 @@
+import org.springframework.web.bind.annotation.RequestParam
+import java.util.UUID
+
+/**
+ * Defines the restful dataland-backend API regarding (company) data ownership.
+ */
+@RequestMapping("/data-owners")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface DataOwnerApi {
+
+    /**
+     * A method to create a new data ownership relation in dataland
+     * @param companyId the ID of the company to which a new data owner is to be added
+     * @param userId the ID of the user who is to be added as company data owner
+     * @return information about the stored company data ownership relation
+     */
+    @Operation(
+        summary = "Add a new data owner to a company.",
+        description = "A new data owner is added to the existing list for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully added data owner."),
+        ],
+    )
+    @PostMapping(
+        produces = ["application/json"],
+
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun postDataOwner(
+        @RequestParam companyId: UUID,
+        @RequestParam userId: UUID,
+    ):
+        ResponseEntity<CompanyDataOwners>
+
+    /**
+     * A method to retrieve a  data owner information from companies in dataland
+     * @param companyId the ID of the company to which a new data owner is to be added
+     * @return userId of the data owner(s) of a specified company
+     */
+    @Operation(
+        summary = "Retrieve data owner(s) of a company.",
+        description = "Get a list of data owner(s) for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data owner."),
+            ApiResponse(responseCode = "404", description = "The specified company does not exist on Dataland."),
+        ],
+    )
+    @GetMapping(
+        produces = ["application/json"],
+    )
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    fun getDataOwners(
+        @RequestParam companyId: UUID,
+    ):
+        ResponseEntity<List<String>>
+
+    /**
+     * A method to delete a new data ownership relation in dataland
+     * @param companyId the ID of the company to which the data ownership should be removed from
+     * @param userId the ID of the user who is to be removed as company data owner
+     * @return information about the deleted company data ownership relation
+     */
+    @Operation(
+        summary = "Delete a data owner from a specified company.",
+        description = "An existing data owner is deleted from the existing list for the specified company.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully deleted data owner."),
+            ApiResponse(responseCode = "404", description = "Data owner doesn't exist for the specified company."),
```

Comment:
What is returned if the company id does not exist?

Reply:
there's currently no differentiation between the companyId not existing (on Dataland) and the companyId not having any owners and therefore not existing in the data owner repository

Reply:
this should be fixed now

Reply:
Is it now a 404 error for the second case (specified company doesn't exists)

Reply:
fixed now

Reply:
![image](https://github.com/d-fine/Dataland/assets/104447806/80f08be4-105d-45bc-8ad9-32170b943cd9)
The error message for an unknown user is not correct

Reply:
The same is true for the delete endpoint if the user is unknown.
![image](https://github.com/d-fine/Dataland/assets/104447806/750f4163-ae6e-4293-8221-769138a40b2a)

Reply:
Are you sure that this error message really corresponds to the issue of an unknown user ID? The screenshot looks like the format of the userID is not correct (not a UUID)?